coaster-nn/src/frameworks/cuda/mod.rs

```diff
@@ -848,14 +902,13 @@ impl<T> Rnn<T> for Backend<Cuda> where T: Float + DataTypeInfo {
             &sequence_descriptors.cx_desc,
             std::ptr::null(),
             &weight_desc,
-            trans_mut!(weight_mem),
+            trans!(weight_mem),
             &sequence_descriptors.hy_desc,
-            // May need to pass hidden at some point for training.
             std::ptr::null_mut(),
             &sequence_descriptors.cy_desc,
             std::ptr::null_mut(),
             trans_mut!(workspace_mem),
-            *reserve_space.id_c(),
+            trans_mut!(reserve_mem),
```

Comment:
I don't quite get why we need `trans_mut!` here, I'd have expected that a plain call to `reserve_mem.id_c()` which gives you a `DriverFFI::CUdeviceptr` which should be just fine.
https://docs.nvidia.com/deeplearning/sdk/cudnn-api/index.html#cudnnRNNForwardTraining
Note that this is a very simplistic view and I very well might have missed something.

```rust
macro_rules! trans_mut {
    ($mem:ident) => (
        unsafe { ::std::mem::transmute::<u64, *mut ::libc::c_void>(*$mem.id_c()) }
    )
}
```
would do something entirely different.
It would use the same id, dererfence it, which would be the value at the address of the pointer given by `.id_c()`, and then cast that (again!) into an address, which might point anywhere in memory. Please correct me if I am wrong.

Reply:
It's a fair point - and for that case it was because I'd moved from using a CudaDeviceMemory, which has an id_c() available, to a SharedTensor as I was trying to test if replicating the workspace memory was effective. 

Unfortunately moving back to *reserve.id_c() using the CudaDeviceMemory still produces the same error.